core/src/main/java/nl/nn/adapterframework/soap/SoapWrapper.java

```diff
@@ -131,43 +143,74 @@ public void checkForSoapFault(Message responseBody, Throwable nested) throws Sen
 		}
 	}
 
-	public Message getBody(Message message) throws SAXException, TransformerException, IOException  {
-		return getBody(message, false, null, null);
-	}
-
 	public Message getBody(Message message, boolean allowPlainXml, PipeLineSession session, String soapNamespaceSessionKey) throws SAXException, TransformerException, IOException  {
 		message.preserve();
-		Message result = new Message(extractBodySoap11.transform(message.asSource()));
-		if (!Message.isEmpty(result)) {
-			if (session!=null && StringUtils.isNotEmpty(soapNamespaceSessionKey)) {
-				session.put(soapNamespaceSessionKey, SoapVersion.SOAP11.namespace);
-			}
-			return result;
-		}
-		result = new Message(extractBodySoap12.transform(message.asSource()));
-		if (!Message.isEmpty(result)) {
-			if (session!=null && StringUtils.isNotEmpty(soapNamespaceSessionKey)) {
-				session.put(soapNamespaceSessionKey, SoapVersion.SOAP12.namespace);
-			}
-			return result;
+
+		// First try with Soap 1.1 transform pool, when no result, try with Soap 1.2 transform pool
+		String extractedBody = extractMessageWithTransformers(extractBodySoap11, extractBodySoap12, message, session, soapNamespaceSessionKey);
+		if (StringUtils.isNotEmpty(extractedBody)) {
+			return new Message(extractedBody);
 		}
-		if (session!=null && StringUtils.isNotEmpty(soapNamespaceSessionKey)) {
+
+		if (session != null && StringUtils.isNotEmpty(soapNamespaceSessionKey)) {
 			session.put(soapNamespaceSessionKey, SoapVersion.NONE.namespace);
 		}
 		return allowPlainXml ? message : new Message(""); // could replace "" with nullMessage(), but then tests fail.
 	}
 
+	private String extractMessageWithTransformers(TransformerPool transformerS11, TransformerPool transformerS12, Message message, PipeLineSession session, String soapNamespaceSessionKey) throws IOException, TransformerException, SAXException {
+		// If SOAP version is already determined in the session, directly use the SOAP 1.2 transformer
+		SoapVersion soapVersion = getSoapVersionFromSession(session);
+		String extractedMessage;
+		if (soapVersion == SoapVersion.SOAP12) {
+			extractedMessage = transformerS12.transform(message.asSource());
+		} else {
+			extractedMessage = transformerS11.transform(message.asSource());
+			if (StringUtils.isNotEmpty(extractedMessage)) {
+				soapVersion = SoapVersion.SOAP11;
+			} else {
+				extractedMessage = transformerS12.transform(message.asSource());
+				if (StringUtils.isNotEmpty(extractedMessage)) {
+					soapVersion = SoapVersion.SOAP12;
+				}
```

Comment:
Yes, I agree. Good suggestion. Will make it work like this as well, and add unit tests for NONE as well.

Reply:
I just figured out that your suggestion is not possible. Because we're using the method `extractMessageWithTransformers` also for optional parts, such as fault  sections, which don't have to be available anyways. Then we can't conclude it is not SOAP. 
When "NONE" is set on purpose, I'll skip parsing, as you suggested above in the previous comment.

Reply:
I've thought up another improvement for this.. I'll add it in the next commit.